Okta/AppAuth/OIDAuthorizationService.m

```diff
@@ -420,29 +420,33 @@ + (void)discoverServiceConfigurationForDiscoveryURL:(NSURL *)discoveryURL
 
 #pragma mark - Token Endpoint
 
-+ (void)performTokenRequest:(OIDTokenRequest *)request callback:(OIDTokenCallback)callback {
-  [[self class] performTokenRequest:request
-      originalAuthorizationResponse:nil
-                           callback:callback];
++ (void)performTokenRequest:(OIDTokenRequest *)request
+                   delegate:(id<OktaNetworkRequestCustomizationDelegate> _Nullable)delegate
+                   callback:(OIDTokenCallback)callback {
+    [[self class] performTokenRequest:request originalAuthorizationResponse:nil delegate:delegate callback:callback];
 }
 
 + (void)performTokenRequest:(OIDTokenRequest *)request
-    originalAuthorizationResponse:(OIDAuthorizationResponse *_Nullable)authorizationResponse
-                         callback:(OIDTokenCallback)callback {
+originalAuthorizationResponse:(OIDAuthorizationResponse *_Nullable)authorizationResponse
+                   delegate:(id<OktaNetworkRequestCustomizationDelegate> _Nullable)delegate
+                   callback:(OIDTokenCallback)callback {
 
   NSURLRequest *URLRequest = [request URLRequest];
-  
+  if ([delegate conformsToProtocol:@protocol(OktaNetworkRequestCustomizationDelegate)]) {
+    URLRequest = [delegate sendCustomizableRequest:URLRequest];
+  }
   AppAuthRequestTrace(@"Token Request: %@\nHeaders:%@\nHTTPBody: %@",
-                      URLRequest.URL,
-                      URLRequest.allHTTPHeaderFields,
-                      [[NSString alloc] initWithData:URLRequest.HTTPBody
+                      customizedRequest.URL,
+                      customizedRequest.allHTTPHeaderFields,
+                      [[NSString alloc] initWithData:customizedRequest.HTTPBody
                                             encoding:NSUTF8StringEncoding]);
 
   NSURLSession *session = [OIDURLSessionProvider session];
   [[session dataTaskWithRequest:URLRequest
               completionHandler:^(NSData *_Nullable data,
                                   NSURLResponse *_Nullable response,
                                   NSError *_Nullable error) {
+    [delegate didReceiveResponse:response];
```

Comment:
don't you need to check conformsToProtocol for this method also ?

Reply:
I don't think we need to check the conform here since we only pass the response to delegate.

Reply:
ok for objective C it will be no-op in case it is not implemented.